c7n/resources/s3.py

```diff
@@ -3328,21 +3347,46 @@ def process(self, buckets):
                                    futures[future]['Name'])
 
     def process_bucket(self, bucket):
+        default_key_desc = 'Default master key that protects my S3 objects when no other key is defined' # noqa
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
         if not self.data.get('enabled', True):
             s3.delete_bucket_encryption(Bucket=bucket['Name'])
             return
         algo = self.data.get('crypto', 'AES256')
-        config = {'Rules': [
-            {'ApplyServerSideEncryptionByDefault': {
-                'SSEAlgorithm': algo}}
-        ]}
+
+        # bucket key defaults to True for alias/aws/s3 and AES256 (Amazon SSE)
+        # and ignores False values for that crypto
+        bucket_key = self.data.get('bucket-key', False)
```

Comment:
a couple of questions on defaults here, does the service error if the values are set to false with sse256 or service key kms?

given the drastic cost reduction on kms usage this feels like something we should default to true with.

Reply:
there's no service error afaics for setting it to false, it is defaulting to true for sse256 and alias/aws/kms

Reply:
looking at this again you're right, I was mostly concerned about this line here: 
```
When you enable an S3 Bucket Key, the AWS KMS key policy for the CMK must include the kms:Decrypt permission for the calling principal. If the calling principal is in a different account than the AWS KMS CMK, you must also include kms:Decrypt permission in the IAM policy. The call to kms:Decrypt verifies the integrity of the S3 Bucket Key before using it.
```

but key policies default to granting access to the account anyway so it should be fine